lib/relation-definition.js

```diff
@@ -532,37 +532,85 @@ function lookupModel(models, modelName) {
   }
 }
 
-function lookupModelTo(modelFrom, modelTo, params, singularize) {
-  if ('string' === typeof modelTo) {
+/*
+ * @return {Object} modelFrom Instance of the 'from' model
+ * @param {Object|String} modelToRef Reference to Model object to which you are
+ *  creating the relation: model instance, model name, or name of relation to model.
+ * @param {Object} params relation params
+ * @param {Boolean} singularize Whether the modelToRef should be singularized when
+ *  looking-up modelTo
+ * @return {Object} modelTo Instance of the 'to' model
+ */
+function lookupModelTo(modelFrom, modelToRef, params, singularize) {
+  var modelTo;
+
+  if ('string' === typeof modelToRef) {
     var modelToName;
-    params.as = params.as || modelTo;
-    modelTo = params.model || modelTo;
-    if (typeof modelTo === 'string') {
-      modelToName = (singularize ? i8n.singularize(modelTo) : modelTo).toLowerCase();
-      modelTo = lookupModel(modelFrom.dataSource.modelBuilder.models, modelToName) || modelTo;
-    }
+    modelTo = params.model || modelToRef; // modelToRef is modelTo name
+    params.as = params.as || modelToRef;  // modelToRef is relationTo name
+
     if (typeof modelTo === 'string') {
-      modelToName = (singularize ? i8n.singularize(params.as) : params.as).toLowerCase();
-      modelTo = lookupModel(modelFrom.dataSource.modelBuilder.models, modelToName) || modelTo;
```

Comment:
The old logic is so confusing....I feel it's purpose is:
- if modelToRef is null, just return it as null
- if typeof modelToRef === 'string', do something else, e.g. override it by params.model if params.model exist

While old code line 540 and line 544 still confuses me....